src/core/pyproxy.js

```diff
@@ -747,38 +739,26 @@ TEMP_EMJS_HELPER(() => {0, /* Magic, see comment */
       let c_contiguous = !!HEAP32[cur_ptr++];
       let f_contiguous = !!HEAP32[cur_ptr++];
 
-      _PyMem_Free(buffer_struct_ptr);
-
       let format = UTF8ToString(format_ptr);
+      _PyMem_Free(buffer_struct_ptr);
 
       let success = false;
       try {
+        let bigEndian = false;
         if (ArrayType === undefined) {
-          // Try to determine correct type from format.
-          // To understand this code it will be helpful to look at the tables
-          // here: https://docs.python.org/3/library/struct.html#format-strings
-          if (format.includes("e")) {
-            throw new Error("Javascript has no Float16Array.");
-          }
-          let cleaned_format = format;
-          // Normalize same-sized types
-          cleaned_format = cleaned_format.replace(/[spc?]/g, "B");
-          cleaned_format = cleaned_format.replace(/[nl]/g, "i");
-          cleaned_format = cleaned_format.replace(/[NLP]/g, "I");
-          let type_char = cleaned_format[0];
-          ArrayType = type_to_array_map.get(type_char);
-          if (ArrayType === undefined) {
-            if (/[qQ]/.test(type_char)) {
-              throw new Error(
-                  "64 bit integer formats (q and Q) are not supported in browsers without BigInt support. You must pass a type argument.");
-            } else {
-              throw new Error(
-                  "Unrecognized buffer format. You must pass a type argument.");
-            }
-          }
+          [ArrayType, bigEndian] = Module.processBufferFormatString(
+              format,
+              " In this case, you must pass an explicit type argument.");
+        }
+        let alignment =
+            parseInt(ArrayType.name.replace(/[^0-9]/g, "")) / 8 || 1;
+        if (bigEndian && alignment > 1) {
+          throw new Error(
+              "Javascript has no native support for big endian buffers. " +
+              "In this case, you must pass an explicit type argument. " +
```

Comment:
Maybe replace "must" by "can"? Even if they do pass a type argument, I'm not sure what they would do with the result.

Say data is 'float32' big endian. What type are users expected to pass: "i8" and try to manually re-order bytes?

Reply:
Well they "must" if they are going to use this function because otherwise it will throw an error... Perhaps "can" is better as you say.

There are a bunch of different options:
1. They could convert in Python, if it's a numpy array with `.byteswap().newbyteorder()`
2. They could use the `toJs` API
3. They could pass `type="dataview"` and then use `dataview.getFloat32(offset, true)` to read a big endian float
4. They could pass `u8` or `i8` and deal with it manually. For instance, in the half float case despite the lack of a Float16Array, webgl2 supports float16 as a texture format. You are supposed to provide it with an ArrayBufferView populated with float16. You could make a f16 array in Python and pass `u8` as the type argument here then hand the Uint8Array to WebGl which interprets it as an array of float16s.

Reply:
Maybe we could add some of the information from your above comment in the error message?